src/reqs.c

```diff
@@ -1498,6 +1533,19 @@ void handle_connection (int fd)
 
         connptr->upstream_proxy = UPSTREAM_HOST (request->host);
         if (connptr->upstream_proxy != NULL) {
+                if (connptr->upstream_proxy->user)
+                {
+                        char plain[256];
```

Comment:
i think we could do with 2 buffers here, e.g. plain and proxy_auth could use the same buffer